frontend/www/js/omegaup/components/identity/Edit.vue

```diff
@@ -0,0 +1,164 @@
+            <div class="col-md-7 col-sm-7">
+              <select class="form-control"
+                   name="stateId"
+                   v-model="selectedState">
+                <option v-bind:value="state.code.split('-')[1]"
+                        v-for="state in countryStates"
+                        v-if="countryStates.length &gt; 0">
+                  {{ state.name }}
+                </option>
+              </select>
+            </div>
+          </div>
+          <div class="form-group">
+            <label class="col-md-4 col-sm-4 control-label"
+                 for="school">{{ T.profileSchool }}</label>
+            <div class="col-md-7 col-sm-7">
+              <input class="form-control"
+                   name="school"
+                   size="20"
+                   type="text"
+                   v-model="identity.school">
+            </div><input name="schoolId"
+                 type="hidden"
+                 v-bind:value="identity.school_id">
+          </div>
+        </div>
+        <div class="form-group pull-right">
+          <button class="btn btn-primary"
+               type="submit">{{ T.wordsSaveChanges }}</button> <button class="btn btn-secundary"
+               type="reset"
+               v-on:click="onCancel">{{ T.wordsCancel }}</button>
+        </div>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Watch, Prop } from 'vue-property-decorator';
+import { T } from '../../omegaup.js';
+import omegaup from '../../api.js';
+import * as iso3166 from '../../../../third_party/js/iso-3166-2.js/iso3166.min.js';
+
+interface State {
+  code: string;
+  name: string;
+}
+
+@Component
+export default class IdentityEdit extends Vue {
+  @Prop() identity!: omegaup.Identity;
+  @Prop() countries!: iso3166.Country[];
+  @Prop({ default: 'MX' }) selectedCountry!: string;
+  @Prop() selectedState!: string;
+  @Prop() username!: string;
+
+  T = T;
+
+  @Watch('selectedCountry')
+  onPropertyChanged(newContry: string, oldCountry: string) {
+    if (this.identity.country_id == newContry) {
+      this.selectedState = this.identity.state_id;
+    } else {
+      this.selectedState = this.countryStates[0].code.split('-')[1];
+    }
+  }
+
+  get groupName(): string {
+    if (typeof this.identity === 'undefined') {
+      return '';
+    }
+    return `${this.identity.username.split(':')[0]}`;
+  }
+
+  get identityName(): string {
+    if (typeof this.identity === 'undefined') {
+      return '';
+    }
+    return this.identity.username.split(':')[1];
+  }
+  set identityName(username) {
+    this.identity.username = `${this.groupName}:${username}`;
+  }
+
+  get countryStates(): State[] {
+    let countrySelected = iso3166.country(this.selectedCountry);
+    let countryStates = Object.keys(countrySelected.sub).map(
+      (code: string) => ({ code: code, name: countrySelected.sub[code].name }),
+    );
+
+    countryStates.sort((a: State, b: State): any => {
```

Comment:
hay que evitar el uso de `any`, porque eso va en contra de la intención de usar TypeScript en primer lugar :(

```suggestion
    countryStates.sort(
        (a: iso3166.Subdivision, b: iso3166.Subdivision): number =>
            Intl.Collator().compare(a.name, b.name));
```

y de paso puedes usar la versión más concisa del lambda. si le pones llaves, _tienes_ que usar `return`.

Reply:
Borré esa función. Ya no se requieren esos cambios